demos/mqtt/mqtt_demo_basic_tls/mqtt_demo_basic_tls.c

```diff
@@ -176,74 +195,42 @@ static uint16_t globalUnsubscribePacketIdentifier = 0U;
 
 /**
  * @brief Array to keep the outgoing publish messages.
- * These stored outgoing publish messages are used to keep the messages until
- * a successful ack is received.
+ * These stored outgoing publish messages are kept until a successful ack
+ * is received.
  */
 static PublishPackets_t outgoingPublishPackets[ MAX_OUTGOING_PUBLISHES ] = { 0 };
 
-/*-----------------------------------------------------------*/
-
 /**
- * @brief Creates a TCP connection to the MQTT broker as specified by
- * BROKER_ENDPOINT and BROKER_PORT defined at the top of this file.
- *
- * @param[in] pServer Host name of server.
- * @param[in] port Server port.
- * @param[out] pTcpSocket Pointer to TCP socket file descriptor.
- *
- * @return EXIT_FAILURE on failure; EXIT_SUCCESS on success.
+ * @brief The network buffer must remain valid for the lifetime of the MQTT context.
  */
-static int connectToServer( const char * pServer,
-                            uint16_t port,
-                            int * pTcpSocket );
+static uint8_t buffer[ NETWORK_BUFFER_SIZE ];
 
-/**
- * @brief Set up a TLS connection over an existing TCP connection.
- *
- * @param[in] tcpSocket Existing TCP connection.
- * @param[out] pSslContext Pointer to SSL connection context pointer.
- *
- * @return EXIT_FAILURE on failure; EXIT_SUCCESS on success.
- */
-static int tlsSetup( int tcpSocket,
-                     SSL ** pSslContext );
-
-/**
- * @brief The transport send function provided to the MQTT context.
- *
- * @param[in] pSslContext Pointer to SSL context.
- * @param[in] pMessage Data to send.
- * @param[in] bytesToSend Length of data to send.
- *
- * @return Number of bytes sent; negative value on error;
- * 0 for timeout or 0 bytes sent.
- */
-static int32_t transportSend( NetworkContext_t pSslContext,
-                              const void * pMessage,
-                              size_t bytesToSend );
+/*-----------------------------------------------------------*/
 
 /**
- * @brief The transport receive function provided to the MQTT context.
+ * @brief connect to MQTT broker with reconnection retries.
+ * If connection fails, retry is attempted after a timeout.
+ * Timeout value will exponentially increased till maximum
+ * timeout value is reached or the number of attemps are exhausted.
  *
- * @param[in] pSslContext Pointer to SSL context.
- * @param[out] pBuffer Buffer for receiving data.
- * @param[in] bytesToRecv Length of data to be received.
+ * @param[out] pNetworkContext Network context pointer for OpenSSL.
```

Comment:
out or in?

Reply:
I believe it is out because it will call `Openssl_Connect` until successful. I need to update the docstring to say `The output parameter to return the created network context.`.